hazelcast-sql/src/main/java/com/hazelcast/sql/impl/calcite/validate/operators/HazelcastDoubleFunction.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.sql.impl.calcite.validate.operators;
+
+import com.hazelcast.sql.impl.calcite.validate.types.ReplaceUnknownOperandTypeInference;
+import org.apache.calcite.sql.SqlFunction;
+import org.apache.calcite.sql.SqlFunctionCategory;
+import org.apache.calcite.sql.SqlKind;
+import org.apache.calcite.sql.type.OperandTypes;
+import org.apache.calcite.sql.type.ReturnTypes;
+import org.apache.calcite.sql.type.SqlTypeName;
+
+import static com.hazelcast.sql.impl.calcite.validate.types.HazelcastOperandTypes.notAny;
+
+/**
+ * Function that accepts a DOUBLE argument and produces a DOUBLE result.
+ */
+public class HazelcastDoubleFunction extends SqlFunction {
+    public HazelcastDoubleFunction(String name) {
+        super(
+            name,
+            SqlKind.OTHER_FUNCTION,
+            ReturnTypes.DOUBLE_NULLABLE,
+            new ReplaceUnknownOperandTypeInference(SqlTypeName.DOUBLE),
```

Comment:
Looks like a proper way of doing this is:

```
InferTypes.explicit(singletonList(HazelcastTypeFactory.INSTANCE.createSqlType(SqlTypeName.DOUBLE)))
```

This basically tells Calcite that the function accepts only doubles, also there would be no need in the conversion performed in `DoubleFunction.eval`, something like `((Number) value).doubleValue()` should do the trick, plain `(double) value` would not work since conversion between numeric types is not necessarily leading to `CAST` insertion in Calcite.

Reply:
Applied the suggested fix